src/main/java/org/cyclonedx/gradle/SbomBuilder.java

```diff
@@ -0,0 +1,249 @@
+        return new Dependency(ref);
+    }
+
+    private void addComponent(
+            final Set<Component> components, final SbomComponent component, final SbomComponent parentComponent) {
+        if (!component.equals(parentComponent)) {
+            final File artifactFile = component.getArtifactFile().orElse(null);
+            try {
+                components.add(toComponent(component, artifactFile, Component.Type.LIBRARY));
+            } catch (MalformedPackageURLException e) {
+                logger.warn("Error constructing packageUrl for component. Skipping...", e);
+            }
+        }
+    }
+
+    private Component toComponent(final SbomComponent component, final File artifactFile, final Component.Type type)
+            throws MalformedPackageURLException {
+
+        final String packageUrl = DependencyUtils.generatePackageUrl(
+                component.getId(), getQualifiers(component.getId().getType()));
+
+        final Component resultComponent = new Component();
+        resultComponent.setGroup(component.getId().getGroup());
+        resultComponent.setName(component.getId().getName());
+        resultComponent.setVersion(component.getId().getVersion());
+        resultComponent.setType(type);
+        resultComponent.setPurl(packageUrl);
+        resultComponent.setProperties(buildProperties(component));
+        resultComponent.setModified(mavenHelper.isModified(null));
+        resultComponent.setBomRef(packageUrl);
+
+        component.getSbomMetaData().ifPresent(metaData -> {
+            resultComponent.setDescription(metaData.getDescription());
+            resultComponent.setPublisher(metaData.getPublisher());
+            metaData.getExternalReferences().forEach(reference -> {
+                final ExternalReference ref = new ExternalReference();
+                ref.setType(ExternalReference.Type.valueOf(reference.getType()));
+                ref.setUrl(reference.getUrl());
+                resultComponent.addExternalReference(ref);
+            });
+        });
+
+        component.getLicenses().ifPresent(licenses -> {
+            LicenseChoice licenseChoice = mavenHelper.resolveMavenLicenses(licenses);
+            resultComponent.setLicenses(licenseChoice);
+        });
+
+        logger.debug(MESSAGE_CALCULATING_HASHES);
+        if (artifactFile != null) {
+            resultComponent.setHashes(calculateHashes(artifactFile));
+        }
+
+        return resultComponent;
+    }
+
+    private List<Property> buildProperties(final SbomComponent component) {
+
+        final String value = component.getInScopeConfigurations().stream()
+                .map(v -> String.format(
+                        "%s:%s", URLEncoder.encode(v.getProjectName()), URLEncoder.encode(v.getConfigName())))
+                .collect(Collectors.joining(","));
+
+        final Property property = new Property();
+        property.setName("inScopeConfiguration");
```

Comment:
There is [CycloneDX property taxonomy](https://github.com/CycloneDX/cyclonedx-property-taxonomy). This properties must be compliant with it. It means that name must start with `cdx` and also must be registered

Reply:
Created proposal for this feature https://github.com/CycloneDX/cyclonedx-property-taxonomy/issues/106

Reply:
Nice! I have included these.

Reply:
I have removed cdx:maven:package:projectsAndScopes for the time being. Will re-introduce this in a future PR.